src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1542 @@
+                        ref_im.WindowWidth,
+                        ref_im.WindowCenter,
+                        getattr(ref_im, 'WindowCenterWidthExplanation', None),
+                        getattr(ref_im, 'VOILUTFunction', None),
+                    )].append(ref_im)
+                elif has_lut:
+                    # Create a unique identifier for this list of LUTs
+                    lut_info = []
+                    for voi_lut in ref_im.VOILUTSequence:
+                        lut_info.append((
+                            voi_lut.LUTDescriptor[1],
+                            voi_lut.LUTDescriptor[2],
+                            getattr(voi_lut, 'LUTExplanation', None),
+                            voi_lut.LUTData
+                        ))
+                    lut_id = tuple(lut_info)
+                    by_lut[lut_id].append(ref_im)
+
+            softcopy_voi_luts = []
+            for (width, center, exp, func), im_list in by_window.items():
+                if len(im_list) == len(referenced_images):
+                    # All datasets included, no need to include the referenced
+                    # images explicitly
+                    refs_to_include = None
+                else:
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        window_center=center,
+                        window_width=width,
+                        window_explanation=exp,
+                        voi_lut_function=func,
+                        referenced_images=refs_to_include
+                    )
+                )
+
+            for lut_id, im_list in by_lut.items():
+                if len(im_list) == len(referenced_images):
+                    # All datasets included, no need to include the referenced
+                    # images explicitly
+                    refs_to_include = None
+                else:
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                luts = [
+                    LUT(
+                        first_mapped_value=fmv,
+                        lut_data=np.frombuffer(
+                            data,
+                            np.uint8 if ba == 8 else np.uint16
+                        ),
+                        lut_explanation=exp
+                    )
+                    for (fmv, ba, exp, data) in lut_id
+                ]
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        referenced_images=refs_to_include,
+                        luts=luts
+                    )
+                )
+
+            if len(softcopy_voi_luts) > 0:
+                self.SoftcopyVOILUTSequence = softcopy_voi_luts
+
+    @staticmethod
+    def _get_icc_profile(ref_im: Dataset) -> Union[bytes, None]:
```

Comment:
I think this is the way to go for the other types of transformations as well (Modality LUT, VOI LUT, etc.). We probably want a getter function to extract the relevant information from a referenced image instance (or a nested sequence item) and a setter function that adds the information to presentation state instance (or a nested sequence item).